lib/utils/compose_ts.ts

```diff
@@ -44,6 +44,64 @@ import type { DeviceInfo } from './device/api';
 import { getBalenaSdk, getChalk, stripIndent } from './lazy';
 import Logger = require('./logger');
 
+/**
+ * The parseReleaseTagKeysAndValues function returns
+ * releaseTagKeys and releaseTagValues given that the
+ * parameter
+ * @param {string[]} releaseTags
+ * @returns { Object } { releaseTagKeys, releaseTagValues }
+ */
+export function parseReleaseTagKeysAndValues(releaseTags: string[]) {
+	if (releaseTags.length === 0) {
+		return { releaseTagKeys: [], releaseTagValues: [] };
+	}
+
+	const releaseTagKeys = releaseTags.filter((_v, i) => i % 2 === 0);
+	const releaseTagValues = releaseTags.filter((_v, i) => i % 2 === 1);
+
+	releaseTagKeys.forEach((key: string) => {
+		if (key === '') {
+			throw new ExpectedError(`Error: --release-tag keys cannot be empty`);
+		}
+		if (/\s/.test(key)) {
+			throw new ExpectedError(
+				`Error: --release-tag keys cannot contain whitespaces`,
+			);
+		}
+	});
+	if (releaseTagKeys.length !== releaseTagValues.length) {
+		releaseTagValues.push('');
+	}
+	return { releaseTagKeys, releaseTagValues };
+}
+
+/**
+ * The parseReleaseTagKeysAndValues function returns
```

Comment:
This description documents more the `parseReleaseTagKeysAndValues` function than the `applyReleaseTagKeysAndValues` function... Building on my comments for that function, here's a suggested description for `applyReleaseTagKeysAndValues`:
```
/**
 * Use the balena SDK `models.release.tags.set()` method to set release tags
 * for the given release ID. The releaseTagKeys and releaseTagValues arrays
 * must be of the same length; their items map 1-to-1 to form key-value pairs.
 */
export async function applyReleaseTagKeysAndValues(
	sdk: BalenaSDK,
	releaseId: number,
	releaseTagKeys: string[],
	releaseTagValues: string[],
) {
```